### PR TITLE
[DEV-5444][DEV-5447] use reducer and config

### PR DESCRIPTION
**Pull request by joshlacey**
## Checklist before requesting a review
- [ ] My pull request was branched from and targets the test branch
- [ ] I have performed a self-review of my code
- [ ] I have manually tested all packages affected (bonus points for automations)
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] My changes generate no new warnings
- [ ] I have checked to ensure there aren't other open pull requests for the same change

## Did you test your feature in the following environments?
- [ ] Standalone Component
- [ ] Standalone Full Editor
- [ ] CDC Internal Checks

### CODE REVIEW

File: packages/dashboard/src/helpers/filterData.ts [hunk 1]
```diff
@@ -0,0 +1,73 @@
...
+    if (!sharedFilter.parents?.length) {
+      return 1
+    } else {
+      let parent = filters.find(filter => sharedFilter.parents!.includes(filter.key))
+      if (!parent) return 1
+      return 1 + findFilterTier(filters, parent)
+    }
+  }
+
+export const filterData = (filters: SharedFilter[], _data: Object[], filterBehavior) => {
```
Comment (adamdoe) [inline packages/dashboard/src/helpers/filterData.ts:14]: Run prettier to resolve alignment issues. Prettier doesn't run when the `-n` flag is used, so its possible this was bypassed.

File: packages/dashboard/src/helpers/generateValuesForFilter.ts [hunk 2]
```diff
@@ -0,0 +1,28 @@
+import { FilterBehavior } from "../components/Header"
```
Comment (adamdoe) [inline packages/dashboard/src/helpers/generateValuesForFilter.ts:1]: Review alignment.

File: packages/dashboard/src/components/Row.tsx [hunk 3]
```diff
@@ -16,11 +13,13 @@ import FourEightColIcon from '../images/icon-col-4-8.svg'
 import EightFourColIcon from '../images/icon-col-8-4.svg'
```
Comment (adamdoe) [inline packages/dashboard/src/components/Row.tsx:13]: Issues importing svgs

File: packages/dashboard/src/components/Row.tsx [hunk 4]
```diff
@@ -1,10 +1,7 @@
 import React, { useContext, useState } from 'react'
```
Comment (adamdoe) [inline packages/dashboard/src/components/Row.tsx:1]: Issues importing React

**Review by adamdoe — COMMENTED** [review]
Nice work @joshlacey. I don't have a lot of feedback, a few questions and a few nits that are up to you if you want to resolve them. Overall it gets the lgtm stamp of approval. I tested locally by moving charts and maps around and adding filters. Let me know if there's anything you'd like me to look at in further depth. I appreciate your attention to detail and organization here. Well done!

questions:
- Can we resolve the svg import errors in Row.tsx/Icons.tsx/etc.
- Can we resolve the new ts error when importing React?

nits:
- Use of prettier
- Alphabetical ordering for readability on things like props/types/reducers

**Review by adamdoe — COMMENTED** [review]
Nice work @joshlacey. I don't have a lot of feedback, a few questions and a few nits that are up to you if you want to resolve them. Overall it gets the lgtm stamp of approval. I tested locally by moving charts and maps around and adding filters. Let me know if there's anything you'd like me to look at in further depth. I appreciate your attention to detail and organization here. Well done!

questions:
- Can we resolve the svg import errors in Row.tsx/Icons.tsx/etc.
- Can we resolve the new ts error when importing React?

nits:
- Use of prettier
- Alphabetical ordering for readability on things like props/types/reducers

File: packages/dashboard/src/helpers/filterData.ts (hunk 1 above)
Comment (joshlacey) [reply packages/dashboard/src/helpers/filterData.ts:14]: I think prettier might not be working on ts files. let me look into it.

File: packages/dashboard/src/components/Row.tsx (hunk 4 above)
Comment (joshlacey) [reply packages/dashboard/src/components/Row.tsx:1]: This should be fixed.

File: packages/dashboard/src/components/Row.tsx (hunk 3 above)
Comment (joshlacey) [reply packages/dashboard/src/components/Row.tsx:13]: should be fixed.

File: packages/dashboard/src/helpers/filterData.ts (hunk 1 above)
Comment (joshlacey) [reply packages/dashboard/src/helpers/filterData.ts:14]: I think this is fixed however I still had to run the no verify flag as it was throwing this error without it:
`Error [ERR_PACKAGE_PATH_NOT_EXPORTED]: Failed to load plugin 'flowtype' declared
in '--config » eslint-config-react-app': Package subpath './lib/rules/no-unused
-expressions' is not defined by "exports" in C:\Users\pro7\dev\cdc-open-viz\node
_modules\eslint\package.json
    at new NodeError (node:internal/errors:405:5)
    at exportsNotFound (node:internal/modules/esm/resolve:359:10)
    at packageExportsResolve (node:internal/modules/esm/resolve:695:9)
    at resolveExports (node:internal/modules/cjs/loader:567:36)
    at Module._findPath (node:internal/modules/cjs/loader:636:31)
    at Module._resolveFilename (node:internal/modules/cjs/loader:1063:27)
    at Module._load (node:internal/modules/cjs/loader:922:27)
    at Module.require (node:internal/modules/cjs/loader:1143:19)
    at require (node:internal/modules/cjs/helpers:121:18)
    at Object.<anonymous> (C:\Users\pro7\dev\cdc-open-viz\node_modules\eslint-pl
ugin-flowtype\dist\rules\noUnusedExpressions.js:8:51)
husky > pre-commit hook failed (add --no-verify to bypass)`

**Review by adamdoe — APPROVED** [review]
@mpallansch can you do a quick pass on this as well? Thanks!

**Review by mpallansch — APPROVED** [review]

**Pull request merged**